bin/multisig.js

```diff
@@ -0,0 +1,387 @@
+
+    this.client = new MultisigClient({
+      network: network.type,
+      port: network.walletPort,
+      apiKey: this.config.str('api-key'),
+    });
+
+    this.wallet = this.client.wallet(this.config.str('wallet'), this.config.str('token'));
+
+    if (this.config.str('path'))
+      this.path = Path.fromString(this.config.str('path'));
+    else
+      this.path = Path.fromOptions({
+        network: network.type,
+        purpose: this.config.uint('purpose', 44),
+        account: this.config.uint('index', 0),
+        // allow for custom coin paths
+        coin: this.config.uint('coin'),
+      });
+
+    /*
+     * get multisig wallet info
+     */
+    if (this.config.has('get-info')) {
+      const info = await this.wallet.getInfo(true);
+      if (!info)
+        throw new Error('could not fetch wallet info');
+
+      this.logger.info('wallet id: %s', info.id);
+      this.logger.info('initialized: %s', info.initialized);
+      if (info.initialized) {
+        this.logger.info('receive address: %s', info.account.receiveAddress);
+        this.logger.info('balance: %s', info.account.balance.confirmed);
+      }
+      process.exit(0);
+    }
+
+    /*
+     * get proposals
+     */
+    if (this.config.has('get-proposals')) {
+      const proposals = await this.wallet.getProposals(true);
+
+      if (!proposals || proposals.length === 0)
+        this.logger.info('no proposals found');
+
+      for (const proposal of proposals)
+        this.logger.info('\n%o', proposal);
+      process.exit();
+    }
+
+    /*
+     * initialize hardware
+     */
+    this.hardware = Hardware.fromOptions({
+      vendor: this.config.str('vendor'),
+      retry: this.config.bool('retry', true),
+      network: network,
+      logger: this.logger,
+    });
+
+    await this.hardware.initialize();
+
+    /*
+     * create multisig wallet
+     */
+    if (this.config.has('create-wallet')) {
+      const cosignerToken = await generateToken(this.hardware, this.path);
+
+      const hdpubkey = await this.hardware.getPublicKey(this.path);
+
+      const response = await this.client.createWallet(this.config.str('wallet'), {
+        witness: this.config.bool('segwit', false),
+        xpub: hdpubkey.xpubkey(network.type),
+        watchOnly: true,
+        m: this.config.uint('m'),
+        n: this.config.uint('n'),
+        cosignerName: this.config.str('cosigner-name'),
+        cosignerPath: this.path.toString(),
+        cosignerToken: cosignerToken.toString('hex'),
+      });
+
+      this.logger.info('wallet created: %s - %s of %s', response.id, response.m, response.n);
+      this.logger.info('join key: %s', response.joinKey);
+
+      const cosigner = response.cosigners.find(c => c.name === this.config.str('cosigner-name'));
+      this.logger.info('cosigner name/token: %s/%s', cosigner.name, cosigner.token);
+    }
+
+    if (this.config.has('join-wallet')) {
+      const hdpubkey = await this.hardware.getPublicKey(this.path.toList());
```

Comment:
Using a different path method for each call to `hardware.getPublicKey` - should just use one